<launch>
  <include file="$(find pr2_machine)/$(env ROBOT).machine" />
  <node machine="c2" pkg="laser_filters" type="scan_to_cloud_filter_chain"
	output="screen" name="scan_to_cloud_filter_chain_tilt_laser">
    <remap from="scan" to="/tilt_scan"/>
    <remap from="cloud_filtered" to="/tilt_scan_shadow2_filtered"/>
    <param name="target_frame" type="string" value="base_footprint"/>
    <rosparam command="load" file="$(find jsk_pr2_startup)/jsk_pr2_sensors/tilt_shadow_filter.yaml" />
  </node>

  <node pkg="pr2_navigation_self_filter"
	type="self_filter" clear_params="true"
	name="robot_shadow_self_filter" respawn="true" output="screen">
    <!-- The topic for the input cloud -->
    <remap from="cloud_in" to="tilt_scan_shadow2_filtered" />
    <!-- The topic for the output cloud -->
    <remap from="cloud_out" to="tilt_scan_shadow2_self_filtered" />
    <!-- The frame of the sensor used to obtain the data to be
	 filtered; This parameter is optional. If it is not specified,
	 shadow points will be considered outside -->
    <!--param name="sensor_frame" type="string" value="laser_tilt_link" /-->
    <!-- Minimum distance to sensor (for point not to be considered inside)-->
    <param name="subsample_value" type="double" value="0.0"/>
    <!-- The names of the links the sensor can see -->
    <rosparam command="load" file="$(find jsk_pr2_startup)/jsk_pr2_sensors/tilt_self_filter.yaml" />
  </node>

  <node type="laser_scan_assembler" name="tilt_scan_assembler" pkg="laser_assembler"
        respawn="true"
        output="screen">
    <remap from="scan" to="tilt_scan"/>
    <param name="max_scans" type="int" value="400" />
    <param name="fixed_frame" type="string" value="base_link" />
    <param name="ignore_laser_skew" type="bool" value="true" />
    <rosparam command="load" file="$(find jsk_pr2_startup)/jsk_pr2_sensors/tilt_scan_filters.yaml" />
  </node>

  <node pkg="collider" type="collider_node" name="collider_node"
	respawn="true" output="screen" machine="c2">
    <param name="fixed_frame" type="string" value="base_link" />
    <param name="resolution" type="double" value="0.01" />
    <param name="pruning_period" type="int" value="3" />
    <!-- <param name="max_range" type="double" value="2.5" /> -->
    <param name="publish_static_over_dynamic_map" value="true" />
    <param name="sensor_model_hit" value= "0.8" />
    <param name="sensor_model_miss" value= "0.31" />
    <param name="sensor_model_min" value= "0.12" />
    <param name="sensor_model_max" value= "0.95" />

    <remap from="collision_map_out" to="collision_map_occ" />
    <!-- <remap from="point_cloud_out" to="octomap_point_cloud" /> -->
    <remap from="point_cloud_out" to="collision_map_point_cloud2" />

    <param name="origin_x" type="double" value="1.1" />
    <param name="origin_y" type="double" value="0.0" />
    <param name="origin_z" type="double" value="0.0" />

    <param name="dimension_x" type="double" value="1.5" />
    <param name="dimension_y" type="double" value="2.0" />
    <param name="dimension_z" type="double" value="2.0" />

    <rosparam command="load" file="$(find jsk_pr2_startup)/jsk_pr2_sensors/collision_map_sources_laser.yaml"/>
    <rosparam command="load" file="$(find jsk_pr2_startup)/jsk_pr2_sensors/tilt_self_filter.yaml" />
  </node>

  <include file="$(find jsk_pcl_ros)/launch/laserscan_registration.launch">
    <arg name="use_laser_pipeline" value="false" />
  </include>

</launch>
